### PR TITLE
Takeaway weekend challenge

### PR DESCRIPTION
**Pull request by Kntanos**
For code review

### CODE REVIEW

File: lib/menu.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require 'csv'
+
+class Menu < Array
+
+  def initialize
+    load_menu
+  end
+
+  def load_menu
+    CSV.foreach("menu.csv", headers: true, header_converters: :symbol, converters: :numeric) do |row|
```
Comment (blue-luke) [inline lib/menu.rb:10]: I like this way of loading the menu, as it implies that it can be done by someone non-technical and then loaded into the programme, which is probably how it would work in real life

File: .gitignore [hunk 2]
```diff
@@ -3,3 +3,5 @@
 
 # Local cache of Rubocop remote config
 .rubocop-*
+
+.env
```
Comment (blue-luke) [inline .gitignore:7]: I like your use of dotenv and .env to store confidential variables

File: lib/checkout.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+require 'dotenv/load'
+require 'twilio-ruby'
+
+class Checkout
```
Comment (blue-luke) [inline lib/checkout.rb:4]: Perhaps there is a better name for this class, as this class only deals with sending a text message? A later class might deal with actual checkout, which would involve payment.

File: lib/menu.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+require 'csv'
+
+class Menu < Array
+
+  def initialize
```
Comment (blue-luke) [inline lib/menu.rb:5]: I also like how you've split this into an initialize and a further function, it separates responsibilities clearly

File: lib/order.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+require_relative 'menu'
+require_relative 'total'
+
+
+class Order
+  attr_reader :menu, :total
+
+  def initialize(menu = Menu.new, total = Total.new)
```
Comment (blue-luke) [inline lib/order.rb:8]: It looks like you've used dependency injection here, nice!

File: lib/order.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  attr_reader :menu, :total
+
+  def initialize(menu = Menu.new, total = Total.new)
+    @menu = menu
+    @total = total
+  end
+
+  def add_to_order(dish, quantity)
+    menu.each do |item|
+      total.selection << item.merge({:quantity => quantity}) if item.value?(dish)
```
Comment (blue-luke) [inline lib/order.rb:15]: I found it a bit hard to parse this line and figure out what the code is doing. You could consider refactoring it to make it more readable

File: myREADME.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
+Run
+```ruby
+$ ruby -r ./lib/order
+```
+
+The test menu used in development is:
```
Comment (blue-luke) [inline myREADME.md:6]: Mentioning the test menu is useful. I also would've like to have seen more information about your approach to this task, difficulties, etc

File: spec/menu_spec.rb [hunk 8]
```diff
@@ -0,0 +1 @@
+require 'menu'
```
Comment (blue-luke) [inline spec/menu_spec.rb:1]: As you've already mentioned you didn't have time to complete some of the tests.

File: spec/order_spec.rb [hunk 9]
```diff
@@ -0,0 +1,16 @@
+require 'order'
+
+describe Order do
+
+  describe "#initialize" do
+    it "should initialize with a menu array" do
```
Comment (blue-luke) [inline spec/order_spec.rb:6]: I like this test, I think it uses dependency injection well

File: spec/order_spec.rb [hunk 10]
```diff
@@ -0,0 +1,16 @@
+require 'order'
+
+describe Order do
+
+  describe "#initialize" do
+    it "should initialize with a menu array" do
+      menu_double = double :menu, new: Array 
+      order = Order.new
```
Comment (blue-luke) [inline spec/order_spec.rb:8]: I've had another look at this, and I'm not sure you've passed in the double to override the default argument. Wouldn't we need to say?:
order = Order.new(menu_double)

Comment (Kntanos) [reply spec/order_spec.rb:8]: Right... it does look strange, but the test fails any other way I tried. I wonder if it is to do with the dependency injection initializing a menu by default?
